documentation/modules/ROOT/pages/03-demo.adoc

```diff
@@ -7,28 +7,879 @@
+
+    @Autowired
+    public DriverFeeService(EventSender eventSender) {
+        this.eventSender = eventSender;
+    }
+
+    public void calculateDriverFee(UUID rideId, Money transitPrice, Long driverId) {
+        eventSender.send(new CalculateFee(
+            rideId,
+            driverId,
+            transitPrice.toInt()
+        ));
+    }
+}
+
+@Service
+public class CloudEventSender {
+
+  private static final Logger log = LoggerFactory.getLogger(EventSender.class);
+
+  private final KnativeConfig knative;
+  private final List<Into<?>> converters;
+
+  @Autowired
+  CloudEventSender(KnativeConfig knative, List<Into<?>> converters) {
+    this.knative = knative;
+    this.converters = converters;
+  }
+
+  public void send(Object event) {
+    try {
+      unsafeSend(event);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private <T> void unsafeSend(T event) throws IOException {
+    Into<T> convert = (Into<T>) converters.stream()
+        .filter(c -> c.accepts(event))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException(
+          "Cannot find converter for " + event.getClass()));
+    CloudEvent ce = convert.into(event);
+    URL url = knative.getSink();
+    log.info("Publishing event to {} : {}", url, ce);
+    HttpURLConnection http = (HttpURLConnection) url.openConnection();
+    http.setRequestMethod("POST");
+    http.setDoOutput(true);
+    http.setDoInput(true);
+
+    HttpMessageWriter messageWriter = createMessageWriter(http);
+    messageWriter.writeBinary(ce);
+
+    int code = http.getResponseCode();
+    if (code < 200 || code >= 300) {
+      throw new IOException("Unexpected response code " + code);
+    }
+  }
+}
+----
+
+Once again, notice this is just a simple _HTTP_ client doing the _POST_ request, with the body being the JSON representation of the _CloudEvent_.
+
+The last part to see is the _HTTP_ listener on the legacy application side.
+This listener will be responsible for receiving events from _Knative's Event Mesh_ and converting them into our custom event type:
+
+[source,java]
+----
+@RestController
+public class CloudEventReceiver {
+  private static final Logger log = LoggerFactory.getLogger(Receiver.class);
+
+  private final EventsPublisher eventsPublisher;
+  private final List<From<?>> froms;
+
+  @Autowired
+  Receiver(EventsPublisher eventsPublisher, List<From<?>> froms) {
+    this.eventsPublisher = eventsPublisher;
+    this.froms = froms;
+  }
+
+  @PostMapping("/")
+  public void receive(@RequestBody CloudEvent event) {
+    log.info("Received event: {}", event);
+
+    for (From<?> from : froms) {
+      if (from.matches(event)) {
+        Event ev = from.fromCloudEvent(event); // <1>
+        eventsPublisher.publish(ev); // <2>
+        return;
+      }
+    }
+
+    throw new IllegalStateException("No matching event type consumer found");
+  }
+}
+----
+
+<1> We unwrap the _CloudEvent_ into our custom event type
```

Comment:
Id' say domain event. and use something like `DomainEvent` to make the difference